src/daft-core/src/array/ops/list.rs

```diff
@@ -472,6 +479,110 @@ impl ListArray {
 }
 
 impl FixedSizeListArray {
+    // this DaftResult<Map>? or something or Series or what
+
+    pub fn list_value_counts(&self) -> DaftResult<MapArray> {
+        struct IndexRef {
+            index: usize,
+            hash: u64,
+        }
+
+        impl std::hash::Hash for IndexRef {
+            fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+                self.hash.hash(state);
+            }
+        }
+
+        let hashes = self.flat_child.hash(None)?;
+        let is_equal = build_is_equal(
+            self.flat_child.to_arrow().as_ref(),
+            self.flat_child.to_arrow().as_ref(),
+            true, // todo: should nulls and nans be considered equal?
+            true,
+        )?;
+
+        let key_type = self.flat_child.data_type().clone();
+        let count_type = DataType::UInt64;
+
+        let fixed_size = self.fixed_element_len();
+
+
+        let mut map: IndexMap<IndexRef, u64, IdentityBuildHasher> = IndexMap::default();
+
+        let mut booleans = Vec::with_capacity(self.flat_child.len());
+        let mut count_array = Vec::new();
+        let mut offsets = Vec::with_capacity(self.len());
+
+        offsets.push(0);
+
+        for i in 0..self.len() {
+            map.clear();
+            let start_index = i * fixed_size;
+            for j in 0..fixed_size {
+                let index = start_index + j;
+
+                let hash = hashes.get(index).unwrap();
+
+                let entry = map.raw_entry_mut_v1().from_hash(hash, |other| {
+                    is_equal(other.index, index)
+                });
+
+                match entry {
+                    RawEntryMut::Occupied(mut entry) => {
+                        booleans.push(false);
+                        *entry.get_mut() += 1;
+                    }
+                    RawEntryMut::Vacant(vacant) => {
+                        booleans.push(true);
+                        vacant.insert(IndexRef { hash, index }, 1);
+                    }
+                }
+            }
+
+
+            // indexmap so ordered
+            for v in map.values() {
+                count_array.push(*v);
+            }
+
+            offsets.push(count_array.len());
+        }
+
+        let values = UInt64Array::from(("count", count_array)).into_series();
+        let boolean_array = BooleanArray::from(("boolean", booleans.as_slice()));
+
+        let keys = self.flat_child.filter(&boolean_array)?;
+
+
+        let r#type = DataType::Struct(vec![
```

Comment:
nit: usually instances of DataType are called `dtype`. Not a hard rule by any means, but makes things a bit easier grok. 

some common conventions I follow: 

```rs
let arr: ArrayRef = ...
let arr: PrimitiveArray = ...
let dtype: DataType = ...
let dtype: ArrowDataType = ...
let s: Series = ...
```

and in python

```py
df: DataFrame = ...
s: Series = ... 
```

Reply:
ok cool. removed btw